lib/web_ui/lib/src/engine/configuration.dart

```diff
@@ -5,10 +5,17 @@
 /// JavaScript API a Flutter Web application can use to configure the Web
 /// Engine.
 ///
-/// The configuration is a plain JavaScript object set as the
-/// `flutterConfiguration` property of the top-level `window` object.
+/// The configuration is passed from JavaScript to the engine as part of the
+/// bootstrap process, through the `FlutterEngineInitializer.initializeEngine`
+/// JS method, with an (optional) object of type [JsFlutterConfiguration].
 ///
-/// Example:
+/// This library also supports the legacy method of setting a plain JavaScript
+/// object set as the `flutterConfiguration` property of the top-level `window`
+/// object, but that approach is now deprecated and will warn users.
+///
+/// Both methods are **disallowed** to be used at the same time.
+///
+/// Example (Before):
```

Comment:
I'd remove the "before" parts, let's boldly step into the new world! :)

Reply:
I disagree, I'll remove these docs when we clean up the code. What I'm going to do, however, is move the new style to the top (and not call it "after"/"new" or anything), and move the legacy style underneath.